#
# Submitted-By: Robert Schwebel, 2006-11-05
# Committed-By: Robert Schwebel
#
# Error:
#
# "/usr/bin/objcopy" --set-section-flags .debug_str=contents,debug "../../../bin/boost/libs/thread/build/libboost_thread.so/gcc/debug/shared-linkable-true/threading-multi/barrier.o"
#
# /usr/bin/objcopy: Unable to recognise the format of the input file `../../../bin/boost/libs/thread/build/libboost_thread.so/gcc/debug/shared-linkable-true/threading-multi/barrier.o'
#
# Description:
#
# There is no possibility to change objcopy to an architecture specific
# variant. This patch makes OBJCOPY configurable.
#
# State:
#
# to be submitted upstream
#

Index: tools/build/v1/gcc-tools.jam
===================================================================
--- tools/build/v1/gcc-tools.jam.orig
+++ tools/build/v1/gcc-tools.jam
@@ -41,6 +41,10 @@
 # GCC
 #       Similar to GXX, the name by which gcc is invoked for "C"
 #       language targets.
+#
+# OBJCOPY
+#       Similar to GXX, the name by which objcopy is invoked.
+#
 
 # singleton variables...
 set-as-singleton GCC_ROOT_DIRECTORY GCC_BIN_DIRECTORY GCC_INCLUDE_DIRECTORY GCC_STDLIB_DIRECTORY ;
@@ -71,6 +75,7 @@ flags gcc CFLAGS <inlining>off : -fno-in
 flags gcc CFLAGS <inlining>on : -Wno-inline ;
 flags gcc CFLAGS <inlining>full : -finline-functions -Wno-inline ;
 
+flags gcc .OBJCOPY : $(OBJCOPY) ;
 flags gcc .GXX : $(GXX) ;
 flags gcc .GCC : $(GCC) ;
 
